### PR TITLE
SONARIAC-1203 S6867: Wildcards should not be used to define RBAC permissions

### PR DESCRIPTION
**Pull request by jonas-wielage-sonarsource**

### CODE REVIEW

File: iac-extensions/kubernetes/src/test/resources/checks/RBACWildcardCheck/wildcardCheckTestTemplate.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  - apiGroups: [""]
+    resources: ["pods/exec"] # Compliant
+    verbs: ["get"]
+
+  - apiGroups: [""]
+    resources: ["services"] # Compliant
+    verbs: ["create"]
+
+  - apiGroups: [""]
+    verbs: ["create"] # Compliant
```
Comment (navendu-barua-sonarsource) [inline iac-extensions/kubernetes/src/test/resources/checks/RBACWildcardCheck/wildcardCheckTestTemplate.yaml:50]: not sure about this, should there be an empty line at the end?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/RBACWildcardCheck.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+      }));
+  }
+
+  private static boolean containsWildCardItem(BlockObject rule, String listKey) {
+    Predicate<YamlTree> wildcardPredicate = TreePredicates.isEqualTo("*");
+    return rule.list(listKey).getItemIf(wildcardPredicate).findAny().isPresent();
+  }
+
+  void reportOnKey(BlockObject rule, String key) {
```
Comment (navendu-barua-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/RBACWildcardCheck.java:59]: may be we should move this method to `AttributeObject` class like others that checks and reports. There is also an existing method `report(textRange, message)` that can be also be used from the new method

**Review by navendu-barua-sonarsource — APPROVED** [review]
Lgtm, left couple of minor comments.

File: iac-extensions/kubernetes/src/test/resources/checks/RBACWildcardCheck/wildcardCheckTestTemplate.yaml (hunk 1 above)
Comment (jonas-wielage-sonarsource) [reply iac-extensions/kubernetes/src/test/resources/checks/RBACWildcardCheck/wildcardCheckTestTemplate.yaml:50]: Not sure what you mean by this. The test is for when `resources` is missing (should not be the case IRL)

Comment (navendu-barua-sonarsource) [reply iac-extensions/kubernetes/src/test/resources/checks/RBACWildcardCheck/wildcardCheckTestTemplate.yaml:50]: Some coding guidelines states to keep an empty line at the end of every file. Is that something to be followed ?

**Pull request merged**